test/e2e/client_tls_test.go

```diff
@@ -384,6 +313,535 @@ func TestClientTLS(t *testing.T) {
 	}
 }
 
+// TestMTLSWithCRLsCerts includes all the certificates needed for a particular test case of TestMTLSWithCRLs.
+type TestMTLSWithCRLsCerts struct {
+	// CABundle is the complete PEM-encoded list of certificates that will be used both by HAProxy for client
+	// validation, and by openshift-router for CRL distribution points.
+	CABundle []string
+	// CRLs is a map of the PEM-encoded CRLs, indexed by the filename that will be used in the CRL host pod's configmap.
+	CRLs map[string]string
+	// ClientCerts contains maps of the client certificates used to verify that mTLS is working as intended.
+	ClientCerts struct {
+		// Accepted is a map containing the client keys and certificates that should be able to connect to backends
+		// successfully, indexed by a unique name.
+		Accepted map[string]KeyCert
+		// Rejected is a map containing the client keys and certificates that should NOT be able to connect to backends,
+		// indexed by a unique name.
+		Rejected map[string]KeyCert
+	}
+}
+
+// TestMTLSWithCRLs verifies that mTLS works when the client auth chain includes certificate revocation lists (CRLs).
+func TestMTLSWithCRLs(t *testing.T) {
+	t.Parallel()
+	namespaceName := "mtls-with-crls"
+	crlHostName := types.NamespacedName{
+		Name:      "crl-host",
+		Namespace: namespaceName,
+	}
+	testCases := []struct {
+		// Name is the name of the test case.
+		Name string
+		// CreateCerts generates the certificates for the test case. Certificates and CRLs must not have expired at the
+		// time of the run, so they must be generated at runtime.
+		CreateCerts func() TestMTLSWithCRLsCerts
+	}{
+		{
+			// This test case has CA certificates including a CRL distribution point (CDP) for the CRL that they
+			// generate and sign. This is the default way to distribute CRLs according to RFC-5280
+			Name: "subject-crl",
```

Comment:
[from meeting] - better name.

i'm not so sure I understand why "subject" is in the name here, but I could see something like: `crls-included-in-ca-certificates`.